### ISSUE TITLE
[eventlet] erratically deletes a unix socket file on worker shutdown

### ISSUE THREAD

**Issue by MarSoft** [benoitc/gunicorn#1320]
Probably this was already discussed, but I couldn't find.
The problem: when I use `eventlet` worker *and* bind to `unix socket`, then the socket file is removed on worker shutdown. As a result, graceful reloading with `HUP` signal is broken.

## Steps to reproduce
1. Create basic app, like [in this example](http://docs.gunicorn.org/en/stable/run.html).
2. Start `gunicorn` like this:
`gunicorn test:app -k eventlet -b unix:/tmp/test.sock`
3. Check that page is served:
`curl --unix-socket /tmp/test.sock http://localhost/`
4. Try to restart `gunicorn`:
`kill -HUP (pid)`
5. Observe an exception message in Gunicorn's output:

    ```
Exception ignored in: <bound method GreenSocket.__del__ of <eventlet.greenio.base.GreenSocket object at 0x7f74b16ef3c8>>
Traceback (most recent call last):
  File "/home/mars/ownCloud/work/bitscore/env/lib/python3.5/site-packages/eventlet/greenio/base.py", line 230, in __del__
    close()
  File "/home/mars/ownCloud/work/bitscore/env/lib/python3.5/site-packages/gunicorn/sock.py", line 123, in close
    os.unlink(self.cfg_addr)
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/a.sock'
[2016-08-04 02:37:56 +0300] [11179] [INFO] Worker exiting (pid: 11179)
```
6. Observe socket file `/tmp/a.sock` disappeared.

Excerpt from `pip freeze`:

    eventlet==0.19.0
    gunicorn==19.6.0

**Comment by diwu1989** [comment]
see https://github.com/benoitc/gunicorn/pull/1309

**Comment by MarSoft** [comment]
Duplicate of #1298, fix is in #1309. Closing.
